frontend/src/metabase-lib/expression.ts

```diff
@@ -48,3 +51,34 @@ export function expressionClause(
 ): ExpressionClause {
   return ML.expression_clause(operator, options, args);
 }
+
+export function isColumnMetadata(
+  arg: ExpressionArg | ExpressionParts,
+): arg is ColumnMetadata {
+  return typeof arg === "object"; // TODO
+}
+
+export function isStringLiteral(
+  arg: ExpressionArg | ExpressionParts,
+): arg is string {
+  return typeof arg === "string";
+}
+
+export function isNumberLiteral(
+  arg: ExpressionArg | ExpressionParts,
+): arg is number {
+  return typeof arg === "number";
+}
+
+export function isNumberLiteralOrCurrent(
+  arg: ExpressionArg | ExpressionParts,
+): arg is number | "current" {
+  return arg === "current" || isNumberLiteral(arg);
+}
+
+export function isTemporalUnit(
+  arg: ExpressionArg | ExpressionParts,
+): arg is TemporalUnit {
+  const units: ReadonlyArray<string> = TEMPORAL_UNITS;
+  return typeof arg === "string" && units.includes(arg);
+}
```

Comment:
Not sure if we want these helpers to be publicly available (like the `Lib.isNumberLiteralOrCurrent` call outside of `metabase-lib`). Should we only keep them internally for high level "filter parts" implementations?